publication-commons/src/main/java/no/unit/nva/publication/service/impl/DynamoDBPublicationService.java

```diff
@@ -84,15 +84,13 @@ public DynamoDBPublicationService(AmazonDynamoDB client, ObjectMapper objectMapp
 
     @Override
     public Publication createPublication(Publication publication) throws ApiGatewayException {
-        UUID identifier = UUID.randomUUID();
         try {
-            publication.setIdentifier(identifier);
             Item item = publicationToItem(publication);
             table.putItem(item);
         } catch (Exception e) {
             throw new DynamoDBException(ERROR_WRITING_TO_TABLE, e);
         }
-        return getPublication(identifier);
+        return getPublication(publication.getIdentifier());
```

Comment:
This is only a solution while I try to think of a better way to handle internal properties on Publication.